Sources/PackageGraph/ModulesGraph+Loading.swift

```diff
@@ -800,6 +781,91 @@ private func createResolvedPackages(
         }
     }
 
+    // Prebuilts
+    for packageBuilder in packageBuilders {
+        for moduleBuilder in packageBuilder.modules {
```

Comment:
Traversing the transitive dependencies of every module separately could get expensive in densely connected package graphs. Could this do a DFS traversal of the graph to mark all the modules ineligible for prebuilts in a single pass?